### PR TITLE
Added  Andes Thumbnail component

### PR DESCRIPTION
**Pull request by alejoecheguia**
## Description
Added new component to andes library,  AndesThumbnail.

## Affected component
It is a new component.

## RFC Link
If this change was discussed on RFC please paste link here.
https://docs.google.com/document/d/1vLBTdig1liPpl-E6Ycqbrxfx5fsnijasJWmHrv2E2TE/edit

## Screenshots / GIFs
This is a UI library, delight us with some stunning images.
Simulator Screen Shot - iPhone 11 Pro Max - 2020-07-06 at 15 40 47
Simulator Screen Shot - iPhone 11 Pro Max - 2020-07-06 at 15 40 55
Simulator Screen Shot - iPhone 11 Pro Max - 2020-07-06 at 15 41 01
Simulator Screen Shot - iPhone 11 Pro Max - 2020-07-06 at 15 41 07

## Checks
Are you sure you followed all those steps? If so, repeat after me "I solemnly swear that ...":
   - [X] I have coded an example inside the Demo App,
   - [X] I have added proper tests,
   - [X] I have modified the Changelog.md file,
   - [ ] I have updated GitHub wiki,
   - [X] I have the explicit approval of one or more members of the UX Team,
   - [X] I have checked that this code is ObjC compliant.
   - [X] I have checked that all the new public enums conform to AndesEnumStringConvertible.
## Change type
This is easy, let us know what this code is about:
   - [X] This code adds a new component
   - [ ] This code includes improvements to an existent component
   - [ ] This code improves or modifies ONLY the Demo App.

### CODE REVIEW

**Comment by furyGorgoryAdmin** [comment]
### Informative Menu <h2>  ### 

### Use any of these tags in a pull request description or leave a comment ### 
| Tag | Description |
|---|---|
| **[canary]**| Creates a build that can only be deployed with canary strategy. | 
| **[build]** |  Creates version or build from a branch that has been tagged as "no releasable". | 
| **[name:version-name]** |Specify the name of the build. For example: [build] [name:1.2.3-name] or [canary] [name:1.2.3-name].|
| **[major]** | To automatically increase the first position of the version. For example: [build] [major] or [canary] [major].|
| **[minor]** | To automatically increase the second position of the version. For example: [build] [minor] or [canary] [minor]. |
| **[patch]** | To automatically increase the third position of the version. For example: [build] [patch] or [canary] [patch].|  

[For more information about ci-cd](http://docs.furycloud.io/?lang=esp&section=businessassurance)
[For more information about semver](https://semver.org/#summary)

File: LibraryComponents/Classes/AndesThumbnail/Size/AndesThumbnailSize.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+	case size_24
+	case size_32
+	case size_40
+	case size_48
+	case size_56
+	case size_64
+	case size_72
+	case size_80
+
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesThumbnail/Size/AndesThumbnailSize.swift:21]: Estos valores del enum veo que no condicen con los definidos en el RFC { 24, 32, 40, 48, 56, 64, 72, 80 }

File: LibraryComponents/Classes/AndesThumbnail/AndesThumbnail.swift [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+		}
+	}
+
+	@IBInspectable public var image: UIImage! {
+		didSet {
+			self.updateContentView()
+		}
+	}
+
+	@objc public var fallbackIcon: String = "andes_ui_fallback_32"{
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesThumbnail/AndesThumbnail.swift:44]: Acá deberíamos usar el struct [AndesIcons](https://github.com/mercadolibre/fury_andesui-ios/blob/develop/LibraryComponents/Classes/AndesIconsProvider/AndesIcons.swift) donde mapeamos el nombre del icon

File: LibraryComponents/Classes/AndesThumbnail/AndesThumbnail.swift [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+//
+
+import UIKit
+
+@objc public class AndesThumbnail: UIView {
+
+	internal var contentView: AndesThumbnailView!
+
+	@objc public var hierarchy: AndesThumbnailHierarchy = .defaultHierarchy {
+		didSet {
```
Comment (samuelsainz) [inline LibraryComponents/Classes/AndesThumbnail/AndesThumbnail.swift:15]: Puede que esté identado con doble tab? Es por alguna razón?

**Comment by samuelsainz** [comment]
@alejoecheguia hice algunos comentarios por arriba pero lo va a estar revisando en profundidad @nrostanml

File: LibraryComponents/Classes/AndesThumbnail/Size/AndesThumbnailSize.swift (hunk 1 above)
Comment (alejoecheguia) [reply LibraryComponents/Classes/AndesThumbnail/Size/AndesThumbnailSize.swift:21]: Pero no se pueden poner solo números como valores del enum

File: LibraryComponents/Classes/AndesThumbnail/AndesThumbnail.swift (hunk 3 above)
Comment (alejoecheguia) [reply LibraryComponents/Classes/AndesThumbnail/AndesThumbnail.swift:15]: puede ser, lo corrijo.

Comment (nrostanml) [reply LibraryComponents/Classes/AndesThumbnail/AndesThumbnail.swift:15]: tenes swiftlint instalado? deberia haber corrido cuando hiciste el commit :/ (podes instalarlo con brew install swiftlint
)

Comment (alejoecheguia) [reply LibraryComponents/Classes/AndesThumbnail/AndesThumbnail.swift:15]: si lo tengo! y correo cuando hice el commit.. No se porque puede haber quedado mal

Comment (alejoecheguia) [reply LibraryComponents/Classes/AndesThumbnail/AndesThumbnail.swift:15]: image

**Comment by nrostanml** [comment]
image
Tenes content layout guides en los dos xibs de la testapp, tenes que sacarlo (puede que CI pase sin eso porque xcode 11.3.1) te recomiendo que si estas usando xcode 11.3 actualices a 11.5 para que te falle a vos localmente...

File: LibraryComponents/Classes/AndesThumbnail/Size/AndesThumbnailSize.swift [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	public static func keyFor(_ value: AndesThumbnailSize) -> String {
+		switch value {
+			case .size_24: return "SIZE_24"
```
[the remark this answers is not in the document]
Comment (alejoecheguia) [reply LibraryComponents/Classes/AndesThumbnail/Size/AndesThumbnailSize.swift:24]: Me parece bien!

File: LibraryComponents/Classes/AndesThumbnail/AndesThumbnail.swift [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+	- fallbackIcon
+	*/
+	@objc public init( hierarchy: AndesThumbnailHierarchy, type: AndesThumbnailType, size: AndesThumbnailSize, state: AndesThumbnailState, image: UIImage, accentColor: UIColor? = .black, fallbackIcon: String? = "andes_ui_fallback_32") {
+		self.hierarchy = hierarchy
+		self.type = type
+		self.size = size
+		self.state = state
+		self.image = image
+		self.accentColor = accentColor
+		self.fallbackIcon = fallbackIcon!
```
[the remark this answers is not in the document]
Comment (alejoecheguia) [reply LibraryComponents/Classes/AndesThumbnail/AndesThumbnail.swift:75]: De hecho lo tengo que quitar al parámetro porque vimos que n tenia mas sentido tener el fallback!. Cuando suba las correcciones del pr ya lo quito

File: LibraryComponents/Classes/AndesThumbnail/Size/AndesThumbnailSize.swift (hunk 4 above)
[the remark this answers is not in the document]
Comment (alejoecheguia) [reply LibraryComponents/Classes/AndesThumbnail/Size/AndesThumbnailSize.swift]: listo!

**Comment by samuelsainz** [comment]
@alejoecheguia alejo, te animas a acomodar el changelog para la versión `3.6.0` y ya mergeamos esto?

**Comment by alejoecheguia** [comment]
Listo @samuelsainz actualizado el changelog!

**Pull request merged**